Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1487,10 +1489,13 @@ private extension Purchases {
     @objc func applicationWillEnterForeground() {
         Logger.debug(Strings.configure.application_foregrounded)
 
+        guard !self.shouldSkipCacheUpdate() else { return }
```

Comment:
I feel we should extract all the code here to an `updateCachesOnAppForeground` or something like that, and move this there... Just in case we want to do something else on app foreground and we don't want to skip it. NABD though.